recipes/msix/all/patches/1.7/cmake.patch

```diff
@@ -0,0 +1,233 @@
++else()
++message(STATUS "MSIX Packaging SDK version ${PROJECT_VERSION}")
++add_definitions(-DSDK_VERSION="${PROJECT_VERSION}")
++endif()
+ # Validates CMake options.
+ include(msix_options)
+ 
++if(ENABLE_NUGET_PACKAGING)
+ # Configure Package.nuspec
+ if(WIN32)
+     set(MSIX_NUGET_NAME "Microsoft.MSIX.Packaging.Windows")
+@@ -103,6 +113,7 @@ configure_file(${MSIX_PROJECT_ROOT}/Package.nuspec.cmakein ${MSIX_BINARY_ROOT}/P
+ configure_file(${MSIX_PROJECT_ROOT}/Microsoft.MSIX.Packaging.targets ${MSIX_BINARY_ROOT}/build/native/${MSIX_NUGET_NAME}.targets)
+ message(STATUS "Package.Nuspec created")
+ message(STATUS "--------------------------------")
++endif()
+ 
+ # Configure license txt
+ configure_file(${MSIX_PROJECT_ROOT}/LICENSE ${MSIX_BINARY_ROOT}/build/LICENSE)
+@@ -170,9 +181,6 @@ if((MACOS) OR (IOS))
+             message(FATAL_ERROR "Unsupported iOS version: ${IOS_DEPLOYMENT_TARGET}, this project requires at least iOS version 10.0")
+         endif()
+     endif()
+-elseif(AOSP OR LINUX)
+-    # Static libraries must be position independent to be linked with a shared object.
+-    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+ endif()
+ 
+ # Mac needed variables
+@@ -183,10 +191,6 @@ set(CMAKE_MACOSX_RPATH ON)
+ #set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+ #set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+ 
+-add_subdirectory(lib)
+-message(STATUS "libs processed")
+ add_subdirectory(src)
+ message(STATUS "src processed")
+-add_subdirectory(sample)
+-message(STATUS "sample processed")
+ message(STATUS "DONE!")
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index a2655a7b..5d224b64 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -6,4 +6,3 @@ cmake_minimum_required(VERSION 3.8.0 FATAL_ERROR)
+ 
+ add_subdirectory(msix)
+ add_subdirectory(makemsix)
+-add_subdirectory(test)
+diff --git a/src/makemsix/CMakeLists.txt b/src/makemsix/CMakeLists.txt
+index 92b082e6..ebae9789 100644
+--- a/src/makemsix/CMakeLists.txt
++++ b/src/makemsix/CMakeLists.txt
+@@ -24,3 +24,6 @@ target_include_directories(${PROJECT_NAME} PRIVATE ${MSIX_BINARY_ROOT}/src/msix)
+ 
+ add_dependencies(${PROJECT_NAME} msix)
+ target_link_libraries(${PROJECT_NAME} msix)
++
++# define installing rules for target files
++install(TARGETS ${PROJECT_NAME})
+diff --git a/src/msix/CMakeLists.txt b/src/msix/CMakeLists.txt
+index 75d3afdf..e1fb2984 100644
+--- a/src/msix/CMakeLists.txt
++++ b/src/msix/CMakeLists.txt
+@@ -6,6 +6,11 @@ cmake_minimum_required(VERSION 3.8.0 FATAL_ERROR)
+ 
+ project(msix)
+ 
++# Define the library
++add_library(${PROJECT_NAME}
++    msix.cpp
++)
++
+ # Handle exports and flags we need to set
+ list(APPEND MSIX_UNPACK_EXPORTS
+     "UnpackPackage"
+@@ -144,7 +149,7 @@ if(NOT SKIP_BUNDLES)
+     if (WIN32)
+         list(APPEND MsixSrc PAL/Applicability/Win32/Applicability.cpp)
+     elseif(LINUX)
+-        find_package(ICU REQUIRED COMPONENTS uc)
++        target_link_libraries(${PROJECT_NAME} PRIVATE CONAN_PKG::icu)
```

Comment:
Conan V2.0 wont have this namespace target

Reply:
I think this issue should be addressed when Conan 2.0 is released, or can you suggest any solution?

Reply:
Does not make sense to merge something we know will break... we are making an active effort to make the migration easier

 🤔 You can keep the call to find package and use the Conan global targets... here could be `icu::icu`

Reply:
@prince-chrismc I think we will block the usage of `CONAN_PKG`, `CONAN_LIBS` and similar in hook soon

Reply:
That's the best way to fix all the recipes

Reply:
This namespace is provided by the `cmake` generator and as far as I know this generator is the only way to handle cmake settings like `MT`/`MD` passing, so right now the only way to replace this namespace is to use, for example, the `cmake_find_package` generator along with `cmake`. But if the `cmake` generator is removed in Conan 2.0, the recipe will require fix anyway, so does it make sense to add another generator and replace the namespace at the moment?

Reply:
Well the new generators will be close to seamless for the `cmake_find_package_multi` espcisially around the injection of targets. If we stick if the original intended use of the build script it will be more future proof?

and yes there will be changes but it's not obvious what that will be... I know the CCI team has experimented with the new workflow but I personally have not so I am not sure... I have the expectation the cmake wrapper will survive

Reply:
no, CMake wrapper is not expected to survive. all the things `conan_basic_setup` did previously now should be set inside the CMake toolchain file generated by conan.
it should enable more seamless and less intrusive integration.
for CCI recipes, it means few simplifications:
- removal of wrapping `CMakeLists.txt` if any
- removal of any patches/replace_in_files doing `CMAKE_SOURCE_DIR`/`CMAKE_CURRENT_SOURCE_DIR` adjustments - now recipe doesn't change project root
- removal of any patches/replace_in_files doing insertions of `conan_basic_setup`, shouldn't be necessary

probably, it will have more implications on recipes, but at least these should be welcomed changes IMO
the workflow will change only with `CMakeDeps`/`CMakeToolchain` generators, for now you can keep doing that you doing